packages/test/test-end-to-end-tests/src/test/gc/gcStats.spec.ts

```diff
@@ -61,19 +74,39 @@ describeCompat("Garbage Collection Stats", "NoCompat", (getTestObjectProvider) =
 
 	beforeEach(async function () {
 		provider = getTestObjectProvider({ syncSummarizer: true });
-		// These tests validate the GC stats in summary by calling summarize directly on the container runtime.
+		// These tests validate the GC stats in summary by calling summarize directly on the mainContainer runtime.
 		// They do not post these summaries or download them. So, it doesn't need to run against real services.
 		if (provider.driver.type !== "local") {
 			this.skip();
 		}
-		container = await provider.makeTestContainer(defaultGCConfig);
-		mainDataStore = (await container.getEntryPoint()) as ITestDataObject;
-		containerRuntime = mainDataStore._context.containerRuntime as ContainerRuntime;
-		await waitForContainerConnection(container);
+		settings = {};
+		settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+		const testContainerConfig: ITestContainerConfig = {
+			runtimeOptions: {
+				summaryOptions: {
+					summaryConfigOverrides: {
+						state: "disabled",
+					},
+				},
+				gcOptions,
+			},
+			loaderProps: { configProvider: mockConfigProvider(settings) },
+		};
+		mainContainer = await provider.makeTestContainer(testContainerConfig);
+		mainDataStore = (await mainContainer.getEntryPoint()) as ITestDataObject;
+		await waitForContainerConnection(mainContainer);
+
+		// Create a second summarizer for running GC and summarizing so that it doesn't summarize local changes.
+		const summarizerContainer = await provider.loadTestContainer(testContainerConfig);
+		const summarizerDataStore = (await summarizerContainer.getEntryPoint()) as ITestDataObject;
+		summarizerRuntime = summarizerDataStore._context.containerRuntime as ContainerRuntime;
+		summarizerDataStore._root.set("write", "mode");
+		await waitForContainerWriteModeConnectionWrite(summarizerContainer);
```

Comment:
Is this necessary?  ensureSynchronized calls throughout should be sufficient, I would think.  Seems fine to leave it since it's more explicit, but wanted to ask.

Reply:
It is needed. I added a comment to explain why.